packages/@uppy/transloadit/src/AssemblyOptions.js

```diff
@@ -29,10 +29,13 @@ function validateParams (params) {
  * compatible object.
  */
 function normalizeAssemblyOptions (file, assemblyOptions) {
-  // eslint-disable-next-line no-param-reassign
-  assemblyOptions.fields = Array.isArray(assemblyOptions.fields)
-    ? Object.fromEntries(assemblyOptions.fields.map((fieldName) => [fieldName, file.meta[fieldName]]))
-    : {}
+  if (Array.isArray(assemblyOptions.fields)) {
+    // eslint-disable-next-line no-param-reassign
+    assemblyOptions.fields = Object.fromEntries(assemblyOptions.fields.map((fieldName) => [fieldName, file.meta[fieldName]]))
+  } else if (assemblyOptions.fields == null) {
+    // eslint-disable-next-line no-param-reassign
+    assemblyOptions.fields = {}
+  }
 
   return assemblyOptions
```

Comment:
Small clean up + getting rid of eslint ignore's

```suggestion
  if (Array.isArray(assemblyOptions.fields)) {
    const fields = Object.fromEntries(assemblyOptions.fields.map((fieldName) => [fieldName, file.meta[fieldName]]))
    return { ...assemblyOptions, fields }
  } 
  return { ...assemblyOptions. fields: {} }
```

Reply:
This wouldn't work either because the code doesn't do anything with the return value. Thinking a bit more about this, I don't think having a function used only once is really useful, especially if it's using slightly unusual patterns as this one. 
Removing the function.